src/app/core/admin/admin-entity-details/admin-entity-details/admin-entity-details.component.html

```diff
@@ -19,32 +19,63 @@
     let-tabIndex="index"
   >
     <div class="flex-column gap-large padding-top-large">
-      @for (
-        componentConfig of item.components;
-        track componentConfig;
-        let i = $index
-      ) {
-        <div class="section-wrapper">
-          <app-admin-section-header
-            [(title)]="componentConfig.title"
-            (remove)="item.components.splice(i, 1)"
-          ></app-admin-section-header>
-          @if (componentConfig.component === "Form") {
-            <app-admin-entity-form
-              [config]="componentConfig.config"
-              (configChange)="componentConfig.config = $event"
-              [entityType]="entityConstructor"
-              uniqueAreaId="{{ tabIndex }}-{{ i }}"
-            >
-            </app-admin-entity-form>
-          } @else {
-            <app-admin-entity-panel-component
-              [config]="componentConfig"
-              [entityType]="entityConstructor"
-            ></app-admin-entity-panel-component>
-          }
-        </div>
-      }
+      <mat-accordion
+        cdkDropList
+        [cdkDropListData]="item.components"
+        cdkDropListOrientation="vertical"
+        (cdkDropListDropped)="onSectionDrop(item, $event)"
+      >
+        @for (
+          componentConfig of item.components;
+          track componentConfig;
+          let i = $index
+        ) {
+          <mat-expansion-panel
+            class="section-wrapper"
+            [expanded]="
+              item.components.length <= 1 ||
+              expandedSections().has(componentConfig)
```

Comment:
are you sure the `has()` function doesn't similarly cause repeated evaluation during change detection, @Abhinegi2 ?